### DISCUSSION TITLE
How to mount encrypted home, prompting for password.

### DISCUSSION THREAD

**Discussion by jbhanks** [zbm-dev/zfsbootmenu#554] [Questions] [answered]
I'm actually not sure if I should configure this in zfsbootmenu or elsewhere, but I have have my `/home/$USER` in an encrypted ZFS dataset. Every time I boot, I have to first log in, and then do `zfs mount -l homepool/HOME/$USER`.
Is there a mechanism for manually supplying passwords in zfsbootmenu? Most of what I have found was about using a password key file, which is not what I want to do. I just want to type my password to decrypt.

To encrypt my pool I used the options  ` -O encryption=aes-256-gcm    -O keyformat=passphrase`. Yes, I realize now that I should have encrypted my dataset instead of my pool, but that shouldn't be the issue here, right?

**Comment by ahesford** [comment, marked as answer]
This is something you will not be able to configure with ZFSBootMenu, because all system state is lost the minute we kexec into the kernel of your boot environment. It's effectively a reboot.

ZFS ships a `pam_zfs_key` module for this purpose. Set your filesystem passphrase equal to your login passphrase, and the module will unlock and mount the filesystem when you log in.

**Reply by jbhanks** [reply]
Thanks, I will look into that.
